src/operator/contrib/adaptive_avg_pooling-inl.h

```diff
@@ -156,5 +161,13 @@ MSHADOW_XINLINE int get_stride(Tensor<xpu, Dim, DType> tensor, int idx) {
 
 }  // namespace op
 }  // namespace mxnet
-
+namespace std {
+template <>
+struct hash<mxnet::op::AdaptiveAvgPoolParam> {
```

Comment:
Maybe it would be worth to comment why here we returning just 0? If I understand correctly only param of this OP determines output size and shape is already part of NDArray passed to AddSign function

Reply:
This function returns "pointer"  at a tuple.  This comment seems to be unnecessary. Done